### PR TITLE
Make network namespace pinning optional

### PR DESCRIPTION
**Pull request by mrunalp**
This allows one to optionally disable the network namespace pinning.

### CODE REVIEW

File: lib/sandbox/sandbox.go [hunk 1]
```diff
@@ -188,7 +189,7 @@ var (
 // New creates and populates a new pod sandbox
 // New sandboxes have no containers, no infra container, and no network namespaces associated with them
 // An infra container must be attached before the sandbox is added to the state
-func New(id, namespace, name, kubeName, logDir string, labels, annotations map[string]string, processLabel, mountLabel string, metadata *pb.PodSandboxMetadata, shmPath, cgroupParent string, privileged, trusted bool, resolvPath, hostname string, portMappings []*hostport.PortMapping) (*Sandbox, error) {
+func New(id, namespace, name, kubeName, logDir string, labels, annotations map[string]string, processLabel, mountLabel string, metadata *pb.PodSandboxMetadata, shmPath, cgroupParent string, privileged, trusted bool, resolvPath, hostname string, portMappings []*hostport.PortMapping, hostNetwork bool) (*Sandbox, error) {
```
Comment (wking) [inline lib/sandbox/sandbox.go]: Since you're carrying the value in `annotations` you might not need the new `hostNetwork` entry here.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@runcom PTAL and get this merged.

**Comment by runcom** [comment]
wait guys, I'm going to test this out with Kenny today, I'm preparing a sys container image. Let's hold this up till we figure the whole issue out.

**Comment by runcom** [comment]
This solves the netns leak, but AWS mounts are still wedged. With this, we fix "drain" but we can't re-provision the pod on another node because the PV is still attached to the old node. We'll keep investigating

**Comment by wking** [comment]
> LGTM, but seeing a function with 18 parameters makes me twitchy. Wish we could get those into a struct or some such instead.

I agree.  Was my "[just get it from `annotations`][1]" suggestion unworkable for some reason?  If we're using `annotaions` (for crash/restart persistence?), it seems like a reasonable was to keep `New`'s arg count down.

[1]: https://github.com/kubernetes-incubator/cri-o/pull/1378#discussion_r171087711

**Pull request merged**